client/home/lib/virtualmachines/components/machineslist/machinedetails.coffee

```diff
@@ -191,9 +193,24 @@ module.exports = class MachineDetails extends React.Component
 
   onClickBuildLog: (e) ->
 
-    machineUid = @props.machine.get 'uid'
-    kd.singletons.router.handleRoute "/Home/build-logs/#{machineUid}"
+    machineUId = @props.machine.get 'uid'
+    path = '/var/log/cloud-init-output.log'
+    kd.singletons.router.handleRoute "/IDE/#{@props.machine.get 'label'}"
+
+    { computeController } = kd.singletons
+    machine    = computeController.findMachineFromMachineUId machineUId
+
+    file = FSHelper.createFileInstance { path, machine }
+
+    return  unless ideApp = envDataProvider.getIDEFromUId machineUId
 
+    ideApp.tailFile {
+      file
+      description : '
+        Your Koding Stack has successfully been initialized. The log here
+        describes each executed step of the Stack creation process.
+      '
+    }
```

Comment:
you can pass the `Machine` object to this method on `ComputeController` https://github.com/koding/koding/blob/master/client/app/lib/providers/computecontroller.coffee#L1197 it'll handle the rest.